js/main.js

```diff
@@ -24,10 +24,112 @@ function submit(event) {
   data.profile.fullName = $fullNameInput.value;
   data.profile.location = $locationInput.value;
   data.profile.bio = $bioTextarea.value;
-  for (var key in data.profile) {
-    localStorage.setItem(key, data.profile[key]);
-  }
+  dataViewSwap('profile');
+  var dataJson = JSON.stringify(data);
+  localStorage.setItem('data-obj', dataJson);
```

Comment:
You should do the update of the localStorage when the page is about to unload, rather than with each submission 👍